sympy/functions/elementary/tests/test_trigonometric.py

```diff
@@ -227,6 +232,12 @@ def _test_extrig(f, i, e):
     # because the other expansions undo the unevaluated Mul
     assert expand_trig(f(Mul(i, 1, evaluate=False))) == e
     assert abs(f(i) - e).n() < 1e-10
+    assert cos(x).rewrite(besselj) == sqrt(pi*x/2)*besselj(-S.Half, x)
+    assert cos(x).rewrite(besselj).subs(x, pi/3).n() == S.Half.n()
+    assert cos(sin(x)).rewrite(besselj).subs(x, 2).n() == \
```

Comment:
It probably makes more sense to write the exact expression here, rather than evaluate it numerically. Note that if you try to rewrite this to a function that it cannot be rewritten to, this type of test will always pass.

(I noted that there were some tests like this earlier, but in general, they do not make sense.)

Reply:
Should I write an expression or not, @oscargus  and @smichr?

Reply:
> Should I write an expression or not, @oscargus and @smichr?

Should @oscargus and @smichr write an expression?

Reply:
The should be like `assert cos(sin(x)).rewrite(besselj) = ...`.

Reply:
Hey there, @oscarbenjamin! I'm a bit confused about something. Can you help me out? I'm figuring out whether I should use expressions or numerical values. I think there may be a conflict. What do you think I should do?

Reply:
There is no conflict. Expressions should be tested.